ExampleApps/SwiftUIExample/SwiftUIExampleTests/UIKitInteropTests.swift

```diff
@@ -44,62 +48,55 @@ final class UIKitInteropTests: XCTestCase, View {
             }
 
             XCTAssertEqual(vc.saveButton?.willRespondToUser, true)
-            XCTAssertEqual(vc?.emailTextField?.willRespondToUser, true)
+            XCTAssertEqual(vc.emailTextField?.willRespondToUser, true)
             vc.emailTextField?.simulateTouch()
-            vc.emailTextField?.simulateTyping(vc?.welcomeLabel?.text)
+            vc.emailTextField?.simulateTyping(vc.welcomeLabel?.text)
             vc.saveButton?.simulateTouch()
 
             self.wait(for: [proceedCalled], timeout: TestConstant.timeout)
         }
-
-        wait(for: [exp], timeout: TestConstant.timeout)
     }
 
-    func testPuttingAUIKitViewInsideASwiftUIWorkflowWithOtherSwiftUIViews() throws {
+    func testPuttingAUIKitViewInsideASwiftUIWorkflowWithOtherSwiftUIViews() async throws {
         struct FR1: View, FlowRepresentable, Inspectable {
             weak var _workflowPointer: AnyFlowRepresentable?
-            var body: some View { EmptyView() }
+            let str: String
+            init(with str: String) {
+                self.str = str
+            }
+            var body: some View { Text(str) }
         }
         let launchArgs = UUID().uuidString
-        let workflowView = WorkflowLauncher(isLaunched: .constant(true), startingArgs: launchArgs) {
-            thenProceed(with: UIKitInteropProgrammaticViewController.self) {
-                thenProceed(with: FR1.self)
+        let launcher = try await MainActor.run {
+            WorkflowLauncher(isLaunched: .constant(true), startingArgs: launchArgs) {
+                thenProceed(with: UIKitInteropProgrammaticViewController.self) {
+                    thenProceed(with: FR1.self)
+                }
             }
         }
-        var vc: UIKitInteropProgrammaticViewController!
+        .hostAndInspect(with: \.inspection)
+        .extractWorkflowItem()
 
-        let exp = ViewHosting.loadView(workflowView).inspection.inspect { workflowLauncher in
-            let wrapper = try workflowLauncher.view(ViewControllerWrapper<UIKitInteropProgrammaticViewController>.self)
+        try await MainActor.run {
+            let wrapper = try launcher.view(ViewControllerWrapper<UIKitInteropProgrammaticViewController>.self)
             let context = unsafeBitCast(FakeContext(), to: UIViewControllerRepresentableContext<ViewControllerWrapper<UIKitInteropProgrammaticViewController>>.self)
-            vc = try wrapper.actualView().makeUIViewController(context: context)
+            let vc = try wrapper.actualView().makeUIViewController(context: context)
             vc.removeFromParent()
             vc.loadOnDevice()
 
             XCTAssertUIViewControllerDisplayed(isInstance: vc)
 
-            let proceedCalled = self.expectation(description: "proceedCalled")
```

Comment:
Because the test was lying before about what it did. You can either check that proceed was called or you can assert the final screen is shown. This test was trying to do both. It passed because the old closure never got called.

This refactor implicitly tests *both* by virtue of the final assert. The screen displays the launch args. Because they were randomized this means we can be certain that proceed worked